x-pack/plugins/ingest_manager/server/routes/agent/acks_handlers.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// handlers that handle events from agents in response to actions received
+
+import { RequestHandler } from 'kibana/server';
+import { TypeOf } from '@kbn/config-schema';
+import { PostAgentAcksRequestSchema } from '../../types/rest_spec';
+import * as APIKeyService from '../../services/api_keys';
+import { AcksService } from '../../services/agents';
+import { AgentEvent } from '../../../common/types/models';
+import { PostAgentAcksResponse } from '../../../common/types/rest_spec';
+
+export const postAgentAcksHandlerBuilder = function(
+  ackService: AcksService
+): RequestHandler<
+  TypeOf<typeof PostAgentAcksRequestSchema.params>,
+  undefined,
+  TypeOf<typeof PostAgentAcksRequestSchema.body>
+> {
+  return async (context, request, response) => {
+    try {
+      const soClient = ackService.getSavedObjectsClientContract(request);
+      const res = APIKeyService.parseApiKey(request.headers);
+      const agent = await ackService.getAgentByAccessAPIKeyId(soClient, res.apiKeyId as string);
+      const agentEvents = request.body.events as AgentEvent[];
+
+      // validate that all events are for the authorized agent obtained from the api key
+      const notAuthorizedAgentEvent = agentEvents.filter(
+        agentEvent => agentEvent.agent_id !== agent.id
+      );
+
+      if (notAuthorizedAgentEvent && notAuthorizedAgentEvent.length > 0) {
+        return response.badRequest({
+          body:
+            'agent events contains events with different agent id from currently authorized agent',
+        });
+      }
+
+      const agentActions = await ackService.acknowledgeAgentActions(soClient, agent, agentEvents);
+
+      if (agentActions.length > 0) {
+        await ackService.saveAgentEvents(soClient, agentEvents);
+      }
+
+      const body: PostAgentAcksResponse = {
+        action: 'acks',
+        success: true,
+        items: [...agentActions],
```

Comment:
Do we want to return the updated actions?

Reply:
I think it's not going to be used by agent so we can remove it

Reply:
Ok, so a simple ack is fine